### PR TITLE
ISPN-6649 Loop through the interfaces to find the first one with IP addresses

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-6649

### CODE REVIEW

File: server/hotrod/src/test/scala/org/infinispan/server/hotrod/HotRodSingleClusteredNonLoopbackTest.scala [hunk 1]
```diff
@@ -38,7 +39,11 @@ class HotRodSingleClusteredNonLoopbackTest extends MultipleCacheManagersTest {
    @Test(enabled=false) // Disable explicitly to avoid TestNG thinking this is a test!!
    override def createBeforeClass() {
       super.createBeforeClass()
-      val iface = findNetworkInterfaces(false).next
+      val ifaces = findNetworkInterfaces(false)
+      var iface: NetworkInterface = null
+      do {
+         iface = ifaces.next
+      } while (!iface.getInetAddresses.hasMoreElements)
```
Comment (danberindei) [inline server/hotrod/src/test/scala/org/infinispan/server/hotrod/HotRodSingleClusteredNonLoopbackTest.scala]: I think you can do `val iface = findNetworkInterfaces(false).filter(_.getInetAddresses.hasMoreElements).next()`

**Comment by tristantarrant** [comment]
Should be even better now @danberindei

**Comment by Sanne** [comment]
I'll merge this - if it passes tests.

**Comment by Sanne** [comment]
```
Results :

Failed tests: 
  HotRod10DistributionTest.testDistributedPutWithTopologyChanges:72->HotRodMultiNodeTest.startClusteredServer:58->HotRodMultiNodeTest.startClusteredServer:68 » IllegalState
  HotRod10ReplicationTest.testReplicatedPutWithTopologyChanges:105->HotRodMultiNodeTest.startClusteredServer:58->HotRodMultiNodeTest.startClusteredServer:68 » IllegalState
  HotRod11DistributionTest>HotRodMultiNodeTest.createBeforeClass:39->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->HotRodMultiNodeTest.createCacheManagers:25 » ServiceConfiguration
  HotRod11LocalCacheTest>HotRodMultiNodeTest.createBeforeClass:39->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->HotRodMultiNodeTest.createCacheManagers:25 » EmbeddedCacheManagerStartup
  HotRod11StorageOnlyNodesTest.testAddingStorageOnlyNode:93 » Timeout Timed out ...
  HotRod12ReplicationTest>HotRodMultiNodeTest.createBeforeClass:39->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->HotRodMultiNodeTest.createCacheManagers:25 » ServiceConfiguration
  HotRodAsymmetricClusterTest>HotRodMultiNodeTest.createBeforeClass:39->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->createCacheManagers:30 NoClassDefFound
  HotRodAuthenticationTest>SingleCacheManagerTest.createBeforeClass:44->HotRodSingleNodeTest.setup:34->SingleCacheManagerTest.setup:31->HotRodSingleNodeTest.createCacheManager:27->HotRodSingleNodeTest.createTestCacheManager:40 » ServiceConfiguration
  HotRodConcurrentTest.testConcurrentPutRequests:37 » IllegalState failed to cre...
  HotRodDefaultCacheTest>SingleCacheManagerTest.createBeforeClass:44->HotRodSingleNodeTest.setup:34->SingleCacheManagerTest.setup:31->HotRodSingleNodeTest.createCacheManager:27->HotRodSingleNodeTest.createTestCacheManager:40 » ServiceConfiguration
  HotRodDistributionTest>HotRodMultiNodeTest.createBeforeClass:42->HotRodMultiNodeTest.startTestHotRodServer:55 » ServiceConfiguration
  HotRodIdleTimeoutTest>SingleCacheManagerTest.createBeforeClass:44->HotRodSingleNodeTest.setup:34->SingleCacheManagerTest.setup:31->HotRodSingleNodeTest.createCacheManager:27->HotRodSingleNodeTest.createTestCacheManager:40 » ServiceConfiguration
  HotRodMarshallingTest>AbstractMarshallingTest.setUp:26 » ServiceConfiguration ...
org.infinispan.server.hotrod.HotRodMergeTest.clearContent(org.infinispan.server.hotrod.HotRodMergeTest)
  Run 1: HotRodMergeTest>MultipleCacheManagersTest.clearContent:98 » IllegalState No ca...
  Run 2: HotRodMergeTest>MultipleCacheManagersTest.clearContent:98 » IllegalState No ca...

  HotRodMergeTest.createBeforeClass:35->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->createCacheManagers:61->MultipleCacheManagersTest.createClusteredCaches:294->MultipleCacheManagersTest.addClusterEnabledCacheManager:180 » ServiceConfiguration
  HotRodMergeTest.destroy:51 NoClassDefFound org/infinispan/server/hotrod/HotRod...
  HotRodProxyTest>HotRodMultiNodeTest.createBeforeClass:39->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->HotRodMultiNodeTest.createCacheManagers:25 » ServiceConfiguration
  HotRodReplicationTest.testPingWithTopologyAwareClient:61 » IllegalState failed...
  HotRodServerTest.testValidateProtocolServerNullProperties:21 » ServiceConfiguration
org.infinispan.server.hotrod.HotRodSniFunctionalTest.afterMethod(org.infinispan.server.hotrod.HotRodSniFunctionalTest)
  Run 1: HotRodSniFunctionalTest.afterMethod:37 NullPointer
  Run 2: PASS
  Run 3: PASS

  HotRodSniFunctionalTest.testServerAndClientWithDefaultSslContext:45 » ServiceConfiguration
  HotRodStatsClusterTest>HotRodMultiNodeTest.createBeforeClass:39->MultipleCacheManagersTest.createBeforeClass:66->MultipleCacheManagersTest.callCreateCacheManagers:72->HotRodMultiNodeTest.createCacheManagers:25 » ServiceConfiguration
  HotRodStatsTest>SingleCacheManagerTest.createBeforeClass:44->HotRodSingleNodeTest.setup:34->SingleCacheManagerTest.setup:31->HotRodSingleNodeTest.createCacheManager:27->createTestCacheManager:22 » ServiceConfiguration
  SkipCacheLoadHotRodTest>SingleCacheManagerTest.createBeforeClass:44->HotRodSingleNodeTest.setup:35->HotRodSingleNodeTest.createStartHotRodServer:42 » IllegalState
  SkipIndexingHotRodTest>SingleCacheManagerTest.createBeforeClass:44->HotRodSingleNodeTest.setup:34->SingleCacheManagerTest.setup:31->HotRodSingleNodeTest.createCacheManager:27->HotRodSingleNodeTest.createTestCacheManager:40 » ServiceConfiguration
  HotRodDistributedEventsTest>AbstractHotRodClusterEventsTest.testEventForwarding:55 » IllegalState
  HotRodDistributedEventsTest>AbstractHotRodClusterEventsTest.testEventReplayAfterAddingListenerInCluster:267 » IllegalState
  HotRodDistributedEventsTest>AbstractHotRodClusterEventsTest.testFailoverSendsEventsForNewContent:131 » IllegalState
  HotRodEventsTest.testNoEventsAfterRemovingListener:154 NoClassDefFound org/inf...
  HotRodFilterEventsTest.testFilteredEventsNoReplay:93 NoClassDefFound org/infin...
  HotRodFilterEventsTest.testFilteredEventsReplay:74 NoClassDefFound org/infinis...
  HotRodFilterEventsTest.testParameterBasedFiltering:50 NoClassDefFound org/infi...
  HotRodReplicatedEventsTest>AbstractHotRodClusterEventsTest.testEventForwarding:55 » IllegalState
  HotRodReplicatedEventsTest>AbstractHotRodClusterEventsTest.testEventReplayAfterAddingListenerInCluster:267 » IllegalState
  HotRodReplicatedEventsTest>AbstractHotRodClusterEventsTest.testFailoverSendsEventsForNewContent:131 » IllegalState
  HotRodReplicatedEventsTest>AbstractHotRodClusterEventsTest.testNoEventReplayAfterAddingListenerInCluster:290 » IllegalState

Tests run: 302, Failures: 36, Errors: 0, Skipped: 61

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

**Comment by tristantarrant** [comment]
Are you testing the same thing ?
Results :

Tests run: 269, Failures: 0, Errors: 0, Skipped: 0

**Comment by Sanne** [comment]
Merged it, my bad: somehow I think my package manager has rest my `/etc/security/limits.conf` back to defaults.

All good once I set the ulimits right again!

**Pull request closed without merging**